internal/bundle/plain/v0/bundle.go

```diff
@@ -0,0 +1,78 @@
+package v0
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"path/filepath"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	apimacherrors "k8s.io/apimachinery/pkg/util/errors"
+	apimachyaml "k8s.io/apimachinery/pkg/util/yaml"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const ManifestsPath = "manifests"
+
+var GVK = schema.GroupVersionKind{Group: "bundles.rukpak.io", Version: "v0", Kind: "Plain"}
+
+type Bundle struct {
+	Objects []client.Object `json:"objects"`
+}
+
+func (b Bundle) Validate() error {
+	var errs []error
+	if len(b.Objects) == 0 {
+		errs = append(errs, fmt.Errorf("found zero objects: bundles of type %q must have at least one object", GVK))
+	}
+	return apimacherrors.NewAggregate(errs)
+}
+
+func LoadFS(bundleFS fs.FS) (*Bundle, error) {
+	var (
+		b   = &Bundle{}
+		err error
+	)
+	b.Objects, err = getObjects(bundleFS)
+	if err != nil {
+		return nil, fmt.Errorf("load objects: %w", err)
+	}
+	if err := b.Validate(); err != nil {
+		return nil, fmt.Errorf("invalid bundle: %w", err)
+	}
+	return b, nil
+}
+
+func getObjects(bundleFS fs.FS) ([]client.Object, error) {
+	var objects []client.Object
+	entries, err := fs.ReadDir(bundleFS, ManifestsPath)
+	if err != nil {
+		return nil, err
+	}
+	for _, e := range entries {
+		if e.IsDir() {
+			continue
```

Comment:
Based on the code, it looks like it's expected for the bundle manifests to live in `/manifests` and include all manifests at the top-level (no nesting). I can update the spec to specify this more explicitly.